base/docs/bootstrap.jl

```diff
@@ -4,7 +4,15 @@ module DocBootstrap
 
 export @doc
 
-const docs = []
+type List
+    head
+end
+type Node
+    data
+    next
+end
```

Comment:
Not a huge deal but I think we can cut this down to a single type, `List(head, tail)`. Then just start with `docs = nothing` and use `node = docs` below.